File: leet_0476_complement.ts
function findComplement(num: number): number {
    // let s = (num >> 0).toString(2) // also works
    // let s = num.toString(2) // also works
    let s = (num >>> 0).toString(2)
    let r = ''
    for (let c of s) {
        r += (c == '0') ? '1' : '0' 
    }
    // console.log(r)
    return parseInt(r, 2);
};

File: leet_0476_complement.cc
class Solution {
public:
    int findComplement(int num) {
        string  mid;
        while (num)
        {
            mid = to_string(num % 2 ^ 1) + mid;
            num /= 2;
        }
        // cout << mid << endl;
        return stoi(mid, 0, 2);
    }
};
